### PR TITLE
refactor role, add tests

### PR DESCRIPTION
**Pull request by notyost**

### CODE REVIEW

File: templates/ecs.json.j2 [hunk 1]
```diff
@@ -2,184 +2,56 @@
...
     "HostAMI" : {
       "Type" : "AWS::EC2::Image::Id"
     },
-    "SubnetID": {
+    "SubnetIDs" : {
       "Type": "List<AWS::EC2::Subnet::Id>",
       "Description": "List of an existing subnet IDs to use for the load balancer and auto scaling group"
     },
-    "DesiredCapacity": {
+    "DesiredCapacity" : {
```
Comment (simplyadrian) [inline templates/ecs.json.j2]: Do we want this white space ?

**Comment by simplyadrian** [comment]
👍 

@sapiensantiquus please have a look as well.

File: templates/ecs.json.j2 (hunk 1 above)
Comment (notyost) [inline templates/ecs.json.j2]: hehe yeah i was just going for consistency -- i may have gone the wrong direction wrt to consistency hehe

**Comment by sapiensantiquus** [comment]
LGTM 👍

**Pull request merged**